### PR TITLE
Push changes

### PR DESCRIPTION
**Pull request by tomczoink**
Updates to Paddy's current PR, in addition to new pages in the realtime and rest sections.

### CODE REVIEW

File: content/general/push/activate-subscribe.textile [hunk 1]
```diff
@@ -373,3 +373,57 @@ blang[android].
...
+  default: LocalDevice
+  ruby:    Ably::Models::LocalDevice
+
+<%= partial 'types/_local_device' %>
+
+h3(#push-channel).
+    default: PushChannel
+    ruby:    Ably::Models::PushChannel
+
+<%= partial 'types/_push_channel' %>
```
Comment (mattheworiordan) [inline content/general/push/activate-subscribe.textile]: I assume you'll need to update these all to use `partial_version` once all the work is merged.

File: content/general/push/publish.textile [hunk 2]
```diff
@@ -141,12 +141,16 @@ channel.publish({ name: 'example', data: 'data', extras: extras }, function(err)
...
-* Devices identified by their native recipient attributes such as their unique @registrationToken@ in the case of GCM, @deviceToken@ in the case of APNS, or @targetUrl@ and @encryptionKey@ in the case of a Web device. This means is particularly useful when migrating to Ably with existing push notification target devices.
+* Devices identified by their native recipient attributes such as their unique @registrationToken@ in the case of GCM, @deviceToken@ in the case of APNS, or @targetUrl@ and @encryptionKey@ in the case of a Web device. This is particularly useful when migrating to Ably with existing push notification target devices.
 
 See the "push admin publish documentation":/general/push/admin#publish for the client library API details, and the "raw push publish REST API documentation":/rest-api#push-publish for information on the underlying direct publishing endpoint used by the client libraries.
 
+Notably, the "push admin":/general/push/admin is only available in our Ruby and JavaScript libraries. If you need to publish to devices with another library, you should use "channel-based publishing":#channel-broadcast.
```
Comment (mattheworiordan) [inline content/general/push/publish.textile]: Sadly, this is not true.  It is supported in pretty much all of our libraries now, PHP, Python, Node, JS, Ruby, iOS, Android, Java etc.  The only libs that don't support it are Go and .NET, and the latter will add support soonish.

File: content/realtime/push.textile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+"Find out more about the push admin API":/general/push/admin.
+
+h2(#platform-support). Platform support
+
+Ably currently offers support for push notifications on the following platforms:
+
+- "Apple Push Notifications":https://developer.apple.com/notifications/ := supported on all mobile devices running iOS and desktop devices running OS X
+- "Firebase Cloud Messaging":https://firebase.google.com/docs/cloud-messaging/ := supported on all Android and iOS devices, although we use FCM exclusively for Android message delivery
+- Experimental "W3C Push API":https://www.w3.org/TR/push-api/ := experimental support for "modern W3C compliant browsers":https://caniuse.com/#feat=push-api (this does not include Apple's Safari browser)
```
Comment (mattheworiordan) [inline content/realtime/push.textile]: Perhaps add a "get in touch" if you want to use the experiemental browser support.

File: content/rest/push.textile [hunk 4]
```diff
@@ -0,0 +1,56 @@
+---
```
Comment (mattheworiordan) [inline content/rest/push.textile]: THis content is duplicated, should we use a partial to avoid these diverging in future?

**Comment by tomczoink** [comment]
Updated per feedback @mattheworiordan, added other languages besides C# and Go to admin. @paddybyers if you could also give a review it'd be much appreciated.

File: content/general/push/_push_intro.textile [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+"Find out more about direct push notification publishing":/general/push/publish#direct-publishing
+
+h2(#activate). Activating a device and receiving notifications
+
+Every device that will receive push notifications must activate itself with the local operating system or framework, and hook into the push notification services that the underlying platform provides. This functionality is platform-specific and can also vary considerably across not just platforms, but also across the push services that operate on those platforms such as GCM and FCM, both of which are available on the Android platform.
+
+The Ably client libraries aim to abstract away this complexity and platform-specific behaviour by providing a consistent API for device activation, maintenance of the device registration, subscription to Ably channels for receiving push notifications, and of course for receiving the native push notifications themselves.
```
Comment (paddybyers) [inline content/general/push/_push_intro.textile]: We don't abstract "receiving the native push notifications themselves". Apps will still use the existing native platform APIs for this.

File: content/general/push/_push_intro.textile [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+"Find out more about device activations and subscriptions":/general/push/activate-subscribe.
+
+h2(#admin). Managing devices and subscriptions
+
+Whilst the realtime client libraries provide APIs for a device to activate itself (via "@client.push@":/general/push/activate-subscribe) and subscribe for push notifications (via "@channel.push@":/general/push/activate-subscribe), those APIs are intentionally limited to actions pertaining to the device it is run on.
+
+A separate and distinct push admin API is additionally provided in our client libraries specifically designed for use by your servers to facilitate managing and delivering push notifications across all of your registered devices. This API, amongst other things, includes features to manage registered devices, channel subscriptions and deliver push notifications directly. Currently the "push admin API":/general/push/admin is available in our JavaScript and Ruby libraries.
```
Comment (paddybyers) [inline content/general/push/_push_intro.textile]: Push admin is supported by JS, Ruby, Java/Android and iOS

File: content/general/push/activate-subscribe.textile [hunk 7]
```diff
@@ -15,9 +15,11 @@ jump_to:
     - Subscribing to push notifications#subscribing
 ---
 
-<%= partial partial_version('general/push/_activate_subscribe_intro') %>
+Every device that will receive push notifications must activate itself with the local operating system or framework, and hook into the push notification services that the underlying platform provides. This functionality is platform-specific and can also vary considerably across not just platforms, but also across the push services that operate on those platforms such as GCM and FCM, both of which are available on the Android platform.
 
-The client libraries also provide a set of admin functionality that is typically used server-side (in a trusted environment) to manage all devices and push notification delivery and subscriptions. You can find out more in the "push admin documentation":admin.
+The Ably client libraries aim to abstract away this complexity and platform-specific behaviour by providing a consistent API for device activation, maintenance of the device registration, subscription to Ably channels for receiving push notifications, and of course for receiving the native push notifications themselves.
```
Comment (paddybyers) [inline content/general/push/activate-subscribe.textile]: "and of course for receiving the native push notifications themselves": we don't do this bit

**Pull request merged**